src/app/App.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import Layout from '../layout/Layout';
+import WelcomePage from '../pages/Welcome/WelcomePage';
+import NotFoundPage from '../pages/NotFound/NotFoundPage';
+import SignInPage from '../pages/SignIn/SignInPage';
+import SignUpPage from '../pages/SignUp/SignUpPage';
+import GraphiQLPage from '../pages/GraphiQL/GraphiQLPage';
+import ProtectedRoute from '../routes/ProtectedRote';
+import { useAuthState } from 'react-firebase-hooks/auth';
+import { auth } from './service/service';
+import { AppRoutes } from '../routes/routeConfig/routeConfig';
+
+const App = () => {
+  const [user, loading] = useAuthState(auth);
+  const [userLoggedIn, setUserLoggedIn] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (!loading) {
+      setUserLoggedIn(Boolean(user));
+      setIsLoading(loading);
+    } else {
+      setIsLoading(loading);
```

Comment:
Seems, setting state twice is redundant   
```suggestion
      setUserLoggedIn(Boolean(user));
      setIsLoading(loading);
```